src/NServiceBus.Core/Features/FeatureActivator.cs

```diff
@@ -59,12 +59,25 @@ public FeatureDiagnosticData[] SetupFeatures(FeatureConfigurationContext feature
 
         public async Task StartFeatures(IServiceProvider builder, IMessageSession session, CancellationToken cancellationToken = default)
         {
+            var startedTaskControllers = new List<FeatureStartupTaskController>();
+
             // sequential starting of startup tasks is intended, introducing concurrency here could break a lot of features.
             foreach (var feature in enabledFeatures.Where(f => f.Feature.IsActive))
             {
                 foreach (var taskController in feature.TaskControllers)
                 {
-                    await taskController.Start(builder, session, cancellationToken).ConfigureAwait(false);
+                    try
+                    {
+                        await taskController.Start(builder, session, cancellationToken).ConfigureAwait(false);
+                    }
+                    catch (Exception)
+                    {
+                        await Task.WhenAll(startedTaskControllers.Select(controller => controller.Stop(cancellationToken))).ConfigureAwait(false);
```

Comment:
There's no need for any song and dance around `Stop` throwing exceptions, because [it doesn't](https://github.com/Particular/NServiceBus/blob/797bc4a73983d4488e4e168292dfbebf27225265/src/NServiceBus.Core/Features/FeatureStartupTaskController.cs#L35-L38) and, should that ever change, `Should_throw_when_feature_task_fails_on_start_and_should_stop_previously_started_tasks_and_should_abort_starting` will fail.